pkg/server/openapi3/harvester_openapi3.yaml

```
@@ -0,0 +1,466 @@
+            $ref: '#/components/schemas/UUID'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Onboard'
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/OnboardResponse'
+        default:
+          $ref: '#/components/responses/Default'
+      security:
+        - harvester_auth: []
+
+  /tenants/{tenantID}/memberships/{membershipID}:
+    # Accept / Denied in bulk?
+    patch:
+      tags:
+        - Memberships
+      summary: Accept/Denied membership requests
+      parameters:
+        - name: tenantID
+          in: path
+          description: Tenant Identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/UUID'
+        - name: membershipID
+          in: path
+          description: ID of the membership
+          required: true
+          schema:
+            $ref: '#/components/schemas/UUID'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/MembershipAoDR'
+        required: true
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Membership'
+        default:
+          $ref: '#/components/responses/Default'
+      security:
+        - harvester_auth: []
+
+  /tenants/{tenantID}/memberships:
+    get:
+      tags:
+        - Memberships
+      summary: Listing the memberships requests based on the query params.
+      parameters:
+        - name: tenantID
+          in: path
+          description: Tenant Identifier
+          required: true
+          schema:
+            $ref: '#/components/schemas/UUID'
+        - name: federationGroupID
+          in: query
+          description: Federation Group ID in which the request was made
+          required: true
+          schema:
+            $ref: '#/components/schemas/UUID'
+        - name: memberID
+          in: query
+          description: Member ID
+          required: true
+          schema:
+            $ref: '#/components/schemas/UUID'
+        - name: status
+          in: query
+          schema:
+            $ref: '#/components/schemas/MembershipStatus'
+          description: Membership Status
+      responses:
+        '200':
+          description: Successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Membership'
+                maxItems: 500
+        default:
+          $ref: '#/components/responses/Default'
+      security:
+        - harvester_auth: []
+
+  /tenants/{tenantID}/member/{memberID}/csr:
```

A CSR is what the Harvester sends, part of the Certificate issuance procces, and it should be a property of request body.

How about naming it something like...
```suggestion
  /tenants/{tenantID}/member/{memberID}/certificates:
```